management-service/services/experiment_service.go

```diff
@@ -555,6 +551,40 @@ func (svc *experimentService) save(exp *models.Experiment) (*models.Experiment,
 	return svc.GetDBRecord(exp.ProjectID, exp.ID)
 }
 
+func (svc *experimentService) filterFieldValues(query *gorm.DB, params ListExperimentsParams) (*gorm.DB, error) {
+	if params.Fields != nil && len(*params.Fields) != 0 {
+		err := validateListExperimentFieldNames(*params.Fields)
+		if err != nil {
+			return nil, err
+		}
+
+		// Stores a map of unique field names which are unique column names to be selected in the db query
+		fieldNamesSet := make(map[string]struct{})
```

Comment:
Why not use `"github.com/golang-collections/collections/set"` for this, which we commonly use elsewhere in the codebase?

Reply:
Ahh, I tried doing that but I realised we can't extract the values contained within the `Set` as a slice of strings because the underlying field (`hash`) containing all the values added to the `Set` is actually a private field:

```go
type (
	Set struct {
		hash map[interface{}]nothing
	}

	nothing struct{}
)
```

Reply:
Hmm there should be a function to get it out.

Reply:
https://github.com/golang-collections/collections/blob/604e922904d3/set/set.go#L36

Reply:
They seem to all return a `*Set` at most 😕 https://pkg.go.dev/github.com/golang-collections/collections/set

Reply:
An alternative is to run `Do()` with a function that stores all the string values in the set into some slice.